editor/src/components/common/shared-strategies/convert-to-flex-strategy.ts

```diff
@@ -170,18 +188,61 @@ export function convertLayoutToFlexCommands(
   })
 }
 
-function ifElementIsFragmentFirstConvertItToFrame(
+function ifElementIsFragmentLikeFirstConvertItToFrame(
   metadata: ElementInstanceMetadataMap,
-  elementPathTree: ElementPathTrees,
+  allElementProps: AllElementProps,
+  elementPathTrees: ElementPathTrees,
   target: ElementPath,
 ): Array<CanvasCommand> {
-  return convertFragmentToFrame(
+  const type = getElementFragmentLikeType(
     metadata,
-    elementPathTree,
-    {},
+    allElementProps,
+    elementPathTrees,
     target,
-    'convert-even-if-it-has-static-children',
+    'sizeless-div-considered-fragment-like',
   )
+  const elementInstace = MetadataUtils.findElementByElementPath(metadata, target)
+  if (
+    elementInstace == null ||
+    isLeft(elementInstace.element) ||
+    !isJSXElementLike(elementInstace.element.value)
+  ) {
+    return []
+  }
+
+  if (type == 'fragment' || type == 'sizeless-div') {
```

Comment:
We already have a `treatElementAsFragmentLike`, should we use that here?

Reply:
treatElementAsFragmentLike includes conditionals too afaik, so I wouldn't get into that in a fix PR (but we can make that work on a follow-up PR). Also, converting to flex is only allowed with DOM elements, so I'd stay consistent with that